src/libs/Geometry/geom_static.cpp

```diff
@@ -150,21 +162,20 @@ GEOM::GEOM(const char *fname, const char *lightname, GEOM_SERVICE &_srv, long fl
     }
     srv.free(rvb);
     // read vertices
-    auto *_colData = colData;
-    if (lightname != nullptr && flsz / 4 != nvertices)
+    if (colData.size() != nvertices)
     {
-        srv.free(colData);
-        lightname = nullptr;
+        colData.clear();
```

Comment:
Is there any sense reading colData before if we clear it here?
it seems like the following 'for' should just be wrapped into `if (colData.size() == nvertices)`